### ISSUE TITLE
Using variables with subscription

### ISSUE THREAD

**Issue by karibertils** [apollographql/reason-apollo#148]
It is not clear how to use variables with subscriptions.

Below is example where query works with variables, subscription works without variables.

```
let component = ReasonReact.statelessComponent("Example");

module GetArticle = [%graphql
  {|
  query getArticle($id: Int!) {
    article: articles_by_pk(id: $id) {
      id
      title
    }
  }
|}
];

module GetArticle2 = [%graphql
  {|
  subscription getArticle($id: Int!) {
    article: articles_by_pk(id: $id) {
      id
      title
    }
  }
|}
];

module GetArticleQuery = ReasonApollo.CreateQuery(GetArticle);
module GetArticleSub = ReasonApollo.CreateSubscription(GetArticle2);

let make = _children => {
  ...component,
  render: _self => {
    let articleQuery = GetArticle.make(~id=10, ());

    <div>
      <GetArticleQuery variables=articleQuery##variables>
        ...{({result}) => <div />}
      </GetArticleQuery>
      <GetArticleSub> ...{({data}) => <div />} </GetArticleSub>
      <GetArticleSub variables=articleQuery##variables>
        ...{({data}) => <div />}
      </GetArticleSub>
    </div>;
  },
};
```

Subscription does not work with variables.
```
  We've found a bug for you!
  /home/kb/git/easy/assets/src/pages/article/Article.re 47:12-56
  
  45 ┆   </GetArticleSub>
  46 ┆   <GetArticleSub variables=articleQuery##variables>
  47 ┆     ...{({data}) => <div />}
  48 ┆   </GetArticleSub>
  49 ┆ </div>;
  
  The function applied to this argument has type
    (~children: GetArticleSub.renderPropObj => ReasonReact.reactElement) =>
    ReasonReact.componentSpec(ReasonReact.stateless, ReasonReact.stateless,
                               ReasonReact.noRetainedProps,
                               ReasonReact.noRetainedProps,
                               ReasonReact.actionless)
This argument cannot be applied without label
  
>>>> Finish compiling(exit: 1)
```

**Comment by Gregoirevda** [comment]
Did you try with `result` ?

https://github.com/apollographql/reason-apollo/blob/master/examples/realtime-chat-application/src/NewMessageNotification.re#L24

**Comment by karibertils** [comment]
yes i tried it. same error

```
  We've found a bug for you!
  /home/kb/git/easy/assets/src/pages/article/Article.re 29:12-40:12

  27 ┆ <div className=css##main>
  28 ┆   <GetArticleSubscription variables=articleQuery##variables>
  29 ┆     ...{
  30 ┆          ({result}) =>
   . ┆ ...
  39 ┆            }
  40 ┆        }
  41 ┆   </GetArticleSubscription>
  42 ┆ </div>;

  The function applied to this argument has type
    (~children: GetArticleSubscription.renderPropObj =>
                ReasonReact.reactElement) =>
    ReasonReact.componentSpec(ReasonReact.stateless, ReasonReact.stateless,
                               ReasonReact.noRetainedProps,
                               ReasonReact.noRetainedProps,
                               ReasonReact.actionless)
This argument cannot be applied without label

>>>> Finish compiling(exit: 1)
```

**Comment by terjebra** [comment]
I also got the same error as you @karibertils, when trying to use a subscription with variables. To my knowledge this is caused by the children argument which is a labeled argument in **ReasonApolloSubscription.re** module,  but not in **ReasonApolloSubQuery.re**, which works as expected. 

I removed the label from the children argument in **ReasonApolloSubscription.re**  (thus same as in **ReasonApolloSubQuery.re**)  and the code compiled with no errors.

Maybe  @Gregoirevda could shed some light on this matter?
